Source/DoxygenTranslator/src/PyDocConverter.cpp

```diff
@@ -0,0 +1,932 @@
+
+  // Ctor takes the output to determine the current indent and to remove the
+  // extra indent added to it in the dtor and the variable containing the indent
+  // to use, which must be used after every new line by the code actually
+  // updating the output.
+  explicit IndentGuard(string& output, string& indent) :
+    m_output(output),
+    m_indent(indent)
+  {
+    const size_t lastNonSpace = m_output.find_last_not_of(' ');
+    if (lastNonSpace == string::npos) {
+      m_firstLineIndent = m_output.length();
+    } else if (m_output[lastNonSpace] == '\n') {
+      m_firstLineIndent = m_output.length() - (lastNonSpace + 1);
+    } else {
+      m_firstLineIndent = 0;
+    }
+
+    // Notice that the indent doesn't include the first line indent because it's
+    // implicit, i.e. it is present in the input and so is copied into the
+    // output anyhow.
+    m_indent = Level();
+  }
+
+  // Get the indent for the first line of the paragraph, which is smaller than
+  // the indent for the subsequent lines.
+  string getFirstLineIndent() const { return string(m_firstLineIndent, ' '); }
+
+  ~IndentGuard()
+  {
+    m_indent.clear();
+
+    // Get rid of possible remaining extra indent, e.g. if there were any trailing
+    // new lines: we shouldn't add the extra indent level to whatever follows
+    // this paragraph.
+    static const size_t lenIndentLevel = strlen(Level());
+    if (m_output.length() > lenIndentLevel) {
+      const size_t start = m_output.length() - lenIndentLevel;
+      if (m_output.compare(start, string::npos, Level()) == 0)
+          m_output.erase(start);
+    }
+  }
+
+private:
+  string& m_output;
+  string& m_indent;
+  unsigned m_firstLineIndent;
+
+  IndentGuard(const IndentGuard&);
+  IndentGuard& operator=(const IndentGuard&);
+};
+
+// Return the indent of the given multiline string, i.e. the maximal number of
+// spaces present in the beginning of all its non-empty lines.
+static size_t determineIndent(const string& s)
+{
+  size_t minIndent = static_cast<size_t>(-1);
+
+  for (size_t lineStart = 0; lineStart < s.length();) {
+    const size_t lineEnd = s.find('\n', lineStart);
+    const size_t firstNonSpace = s.find_first_not_of(' ', lineStart);
+
+    // If inequality doesn't hold, it means that this line contains only spaces
+    // (notice that this works whether lineEnd is valid or string::npos), in
+    // which case it doesn't matter when determining the indent.
+    if (firstNonSpace < lineEnd) {
+      // Here we can be sure firstNonSpace != string::npos.
+      const size_t lineIndent = firstNonSpace - lineStart;
+      if (lineIndent < minIndent)
+        minIndent = lineIndent;
+    }
+
+    if (lineEnd == string::npos)
+      break;
+
+    lineStart = lineEnd + 1;
+  }
+
+  return minIndent;
+}
+
+static void trimWhitespace(string& s)
+{
+  const size_t lastNonSpace = s.find_last_not_of(' ');
+  if (lastNonSpace != string::npos)
+    s.erase(lastNonSpace + 1);
+}
```

Comment:
You're right, but I don't think `s` can ever consist of just spaces here, so changing this shouldn't change anything. I'll still do it, although I'll use `clear()` in `npos` case for clarity even if, again, you're right and calling `erase()` unconditionally would work too.

Thanks!